test-apps/display-test-app/vite.config.ts

```diff
@@ -32,10 +44,29 @@ Object.keys(packageJson.dependencies).forEach((pkgName) => {
         // filter out pkgs that actually dont have assets
         assets.push(assetsPath);
       }
-    } catch {}
+    } catch { }
+  }
+  if (pkgName.startsWith("@itwin") && !itwinPackagesWithoutAliases.includes(pkgName)) {
+    if (pkgName.includes("electron") || pkgName.includes("mobile")) {
+      const fileName = pkgName.replace("@itwin/", "");
+      packageAliases = { ...packageAliases, [`${pkgName}/lib/cjs/${fileName.split("-")[1].charAt(0).toUpperCase()}${fileName.split("-")[1].substring(1)}Frontend`]: findPathToPackage(pkgName) }
+    } else {
+      packageAliases = { ...packageAliases, [pkgName]: findPathToPackage(pkgName) }
+    }
   }
 });
 
+function findPathToPackage(packageName: string): string {
+  if (packageName.includes("core-electron") || packageName.includes("core-mobile")) { //require.resolve does not work for core-electron or core-mobile, so these packages are mutated based on their name.
```

Comment:
if we hardcode the electron and mobile alias, then we dont need all this duplicated logic that you have

all you would need is the part of the else statement